tests/std/tests/P2440R1_ranges_alg_shift_left/test.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+struct int_wrapper {
+    int val                 = 10;
+    constexpr int_wrapper() = default;
+    constexpr int_wrapper(int x) : val{x} {}
+    constexpr int_wrapper(int_wrapper&& that) : val{exchange(that.val, -1)} {}
+    constexpr int_wrapper& operator=(int_wrapper&& that) {
+        val = exchange(that.val, -1);
+        return *this;
+    }
+    auto operator<=>(const int_wrapper&) const = default;
+};
+
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::shift_left(borrowed<false>{}, 1)), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::shift_left(borrowed<true>{}, 1)), ranges::subrange<int*>>);
+
+
+struct instantiator {
+    static constexpr int_wrapper expected_result[2] = {12345, 3333};
+    static constexpr int_wrapper expected_io[5]     = {12345, 3333, 44, -1, -1};
+
+    template <ranges::forward_range Rng>
+        requires permutable<ranges::iterator_t<Rng>>

```

Comment:
We have a few tests that apply constraints in the instantiator, we conventionally do so via `if constexpr (meow)` in the body of call instead of using concept overloading like this. The cost of overload resolution is probably dwarfed by the cost of instantiating the actual test code, so it's probably not important for test performance, but it would be nice to be consistent with the pattern we use elsewhere. (Also applies to the `shift_right` test.)